praetorian_cli/ui/aegis/commands/set.py

```
@@ -17,6 +17,18 @@ def handle_set(menu, args):
         menu.selected_agent = selected_agent
         hostname = selected_agent.hostname
+
+        # In multi-account mode, assume into the agent's account so SDK
+        # calls (asset search, domain lookup, etc.) target the right tenant.
+        if getattr(menu, 'multi_account_mode', False):
+            acct_info = menu.agent_account_map.get(selected_agent.client_id, {})
+            acct_email = acct_info.get('account_email')
+            if acct_email:
+                try:
+                    menu.sdk.accounts.assume_role(acct_email)
+                except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)